kernel/src/fs/fs_resolver.rs

```diff
@@ -85,21 +88,39 @@ impl FsResolver {
         Ok(())
     }
 
-    /// Lookups the target path according to the `fs_path`.
+    /// Lookups the target `Path` according to the `fs_path`.
     ///
     /// Symlinks are always followed.
     pub fn lookup(&self, fs_path: &FsPath) -> Result<Path> {
-        self.lookup_inner(fs_path, true)?.into_path()
+        self.lookup_target(fs_path)?
+            .into_path()
+            .ok_or_else(|| Error::with_message(Errno::EPERM, "the path refers to a pseudo file"))
     }
 
-    /// Lookups the target path according to the `fs_path`.
+    /// Lookups the target `Path` according to the `fs_path`.
     ///
     /// If the last component is a symlink, it will not be followed.
     pub fn lookup_no_follow(&self, fs_path: &FsPath) -> Result<Path> {
-        self.lookup_inner(fs_path, false)?.into_path()
+        self.lookup_target_no_follow(fs_path)?
+            .into_path()
+            .ok_or_else(|| Error::with_message(Errno::EPERM, "the path refers to a pseudo file"))
     }
 
-    /// Lookups the target path according to the `fs_path` and leaves
+    /// Lookups the target `PathOrInode` according to the `fs_path`.
+    ///
+    /// Symlinks are always followed.
+    pub fn lookup_target(&self, fs_path: &FsPath) -> Result<PathOrInode> {
```

Comment:
I cannot think of a perfect name for this. But will `lookup_inode` sound better than the current name? I don't really know what `target` is supposed to mean.